internal/resources/cluster/data_source_cluster.go

```diff
@@ -48,6 +48,13 @@ func dataSourceTMCClusterRead(ctx context.Context, d *schema.ResourceData, m int
 				return false, nil
 			}
 
+			if clienterrors.IsUnauthorizedError(err) {
+				md, _ := authctx.GetUserAuthCtx(&config)
```

Comment:
Can we have a function defined in the terms of `refreshUserAuthCtx` and this function has these lines 

Func RefreshUserAuthCtx(){
md, _ := authctx.GetUserAuthCtx(&config)
 				for key, value := range md {
 					config.TMCConnection.Headers.Set(key, value)
 				}
}

And lets not expose GetUserAuthCtx

Reply:
infact you can define functional parameters to enhance the refresh usecase. 

RefreshUserAuthCtx(withRefreshOnAuthError(err))
RefreshUserAuthCtx(withForceRefresh())

This would mean that the error check is also pushed into the refresh logic

Reply:
I've updated the code according to the comment, PTAL. I'll try to get Vasundhara or Shobha's help on the test, as right now the acceptance test seems to detach the cluster prematurely before the cluster come up to the ready state. I'll post my update once I complete the test.